### DISCUSSION TITLE
iOS 16/17 .destination not working after view is hidden/shown (menu) but it works on iOS 18

### DISCUSSION THREAD

**Discussion by jerometonnelier** [pointfreeco/swift-composable-architecture#3502] [General]
### Description

Hi everyone,

weird bug here.
I've got a SwiftUI view that represents a Menu (embed inside a UIViewController since the rest of the project is UIKit based) and used along with [SideMenu](https://github.com/jonkykong/SideMenu/tree/master). 
I used one of the latest TCA implementation with @Perception.Bindable var store: StoreOf<MobileMenuFeature>, @Reducer reducers, @ObservableState states, @Presents and CasePath macros to push some views onto a NavigationStack.

With iOS 18, everything works fine, I can push some views on the SwiftUI Stack, show the UIKit Rootview of the UIKit core part. 
But with iOS17, I get this issue : when I first launch the app, everything works fine the first time I show the menu. I can navigate inside, no worries. But if I close the menu and open it again, no more view is pushed onto the stack (or dismissed for that matter. The action is triggered, the @Presents variable is filled, but I never enter the .destination modifier.

Here is the one of the modifiers
```
.navigationDestination(
                                    store: self.store.scope(
                                        state: \.$taximeterSubMenu,
                                        action: \.taximeterSubMenuAction)) { subStore in
                                            WithPerceptionTracking {
                                                TaximeterMenuView(store: subStore)
                                                    .onAppear { handleNavBar() }
                                            }
                                        }
```
the Case Path macro
```
static let taximeterSubMenuCasePath = AnyCasePath<Action, PresentationAction<TaximeterMenuFeature.Action>> (
            embed: { .taximeterSubMenuAction($0) },
            extract: { if case let .taximeterSubMenuAction(value) = $0 { return value } else { return nil } }
        )
```

Is there some breakpoint I can make or logs to activate to try to see what happens ? 
I'll check with the company if it's possible to send the source code, that would be easier.

Thanks

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### The Composable Architecture version information

1.15.2

### Destination operating system

iOS 17

### Xcode version information

Xcode 16.2

### Swift Compiler version information

```shell
swift-driver version: 1.115 Apple Swift version 6.0.2 (swiftlang-6.0.2.1.2 clang-1600.0.26.4)
Target: arm64-apple-macosx15.0
```

**Comment by jerometonnelier** [comment]
fyi, I tried to target iOS 17 and remove all `WithPerceptionTracking`, I get the same result.
I also added some breakpoints inside the 
```
        .ifLet(\.$taximeterSubMenu, action: Action.taximeterSubMenuCasePath) {
            TaximeterMenuFeature()
        }
```
and 
```
        static let taximeterSubMenuCasePath = AnyCasePath<Action, PresentationAction<TaximeterMenuFeature.Action>> (
            embed: { .taximeterSubMenuAction($0) },
            extract: { if case let .taximeterSubMenuAction(value) = $0 {
                return value
            } else {
                return nil
            } }
        )
```
I do enter several times inside both, the difference being that the first time, I eventually enter the `return value` while all other times, I only get the `return nil`. I hope that can help

**Comment by OguzYuuksel** [comment]
Hi @jerometonnelier,
Could you provide a sample project repository where the issue can be reproduced so that we can investigate further?
It might be related to the an open PR: https://github.com/pointfreeco/swift-composable-architecture/pull/3464
Could you try that branch just to confirm its identity issue?

**Comment by mbrandonw** [comment]
Hi @jerometonnelier, as @OguzYuuksel mentioned, a reproducing project is the best way for someone to look into this.

However, the fact that this works in iOS 18 and not 16/17 leads to me to believe it is a SwiftUI bug and probably not related to this library. Because of that I am going to convert this to a discussion for the time being and please feel free to keep the conversation going over there.

**Comment by jerometonnelier** [comment]
Thanks to both of you, no worries for the discussion. I'll try to make a small project and see if I can reproduce it since my client does not want his code to be shared :-/  I'll keep you posted asap. Thanks again
